### PR TITLE
Add hex dump of any buffer from parser errors

### PR DESCRIPTION
**Pull request by jcorbin**

### CODE REVIEW

**Comment by jcorbin** [comment]
We may not want this as is since it jams a pretty printed string into a log message, but the utility of it is invaluable for debugging.

File: index.js [hunk 1]
```diff
@@ -396,7 +397,12 @@ require('util').inherits(TChannelConnection, require('events').EventEmitter);
 
 TChannelConnection.prototype.onParserError = function onParserError(err) {
     var self = this;
-    self.channel.logger.error('tchannel parse error', {
+    var mess = 'tchannel parse error';
+    if (err.buffer) {
+        mess += '\n' + hexer(err.buffer);
```
Comment (Raynos) [inline index.js]: consider adding a second `logger.debug()` message for the buffer.

Remember that errors go to sentry.

File: index.js [hunk 2]
```diff
@@ -28,6 +28,7 @@ var globalNow = Date.now;
 var globalRandom = Math.random;
 var net = require('net');
 var inspect = require('util').inspect;
+var hexer = require('hexer');
```
Comment (kriskowal) [inline index.js]: reviewed hexer. legitimate.

**Comment by jcorbin** [comment]
obsolete, will be redoing as needed

**Pull request closed without merging**